File: libs/domain-models/analysis.ts
export type ChangeType = 'api' | 'schema' | 'infra' | 'config';
export type RiskLevel = 'low' | 'medium' | 'high' | 'critical';

export interface ChangeProposal {
  serviceId: string;
  changeType: ChangeType;
  title: string;
  description: string;
}

export interface ImpactedService {
  service: string;
  reason: string;
  score: number;
}

export interface ImpactResult {
  riskScore: number;
  riskLevel: RiskLevel;
  affectedServices: ImpactedService[];
  recommendedActions: string[];
}

File: libs/domain-models/analysis.py
from typing import Literal
from pydantic import BaseModel, Field

ChangeType = Literal['api', 'schema', 'infra', 'config']
RiskLevel = Literal['low', 'medium', 'high', 'critical']


class ChangeProposal(BaseModel):
    service_id: str = Field(..., examples=['auth-gateway'])
    change_type: ChangeType = Field(..., examples=['api'])
    title: str = Field(..., examples=['Add OAuth fallback'])
    description: str = Field(..., examples=['Add fallback for external OIDC outage'])


class ImpactedService(BaseModel):
    service: str
    reason: str
    score: int


class ImpactResult(BaseModel):
    risk_score: int = Field(..., ge=0, le=100)
    risk_level: RiskLevel
    affected_services: list[ImpactedService]
    recommended_actions: list[str]
